src/mat.h

```diff
@@ -39,6 +39,14 @@
 #endif // __ANDROID_API__ >= 9
 #endif // NCNN_PIXEL
 
+#if !defined(HEDLEY_MSVC_VERSION) || HEDLEY_MSVC_VERSION_CHECK(19, 20, 0)
+#define NCNN_OMP_SIMD            HEDLEY_PRAGMA(omp simd)
+#define NCNN_OMP_SIMD_EX(Extra)  HEDLEY_PRAGMA(omp simd Extra)
+#else
+#define NCNN_OMP_SIMD
+#define NCNN_OMP_SIMD_EX(Extra)
+#endif
+
```

Comment:
How is the actual speedup benefit of omp simd clause ?
We use explicit simd intrinsic and inline assembly for most critical code paths, I don't think the compiler could do better.

Reply:
There are a lot of generic code still present in here, and for simple cases auto-vectorization is less error-prone. This also speeds up debug-mode since it's previously not vectorized. MSVC also specifically tells you why something could not be vectorized when you add this.